src/main/java/bowling/controller/BowlingController.java

```diff
@@ -24,38 +24,22 @@ public BowlingController(InputView inputView, OutputView outputView) {
     }
 
     public void run() {
-        final Player player = new Player(playerName());
-        outputView.printScoreBoard(player);
-
-        for (int i = RoundNumber.MIN; i <= RoundNumber.MAX; i++) {
-            bowl(player, i);
-        }
-    }
-
-    private void bowl(Player player, int i) {
-        final RoundNumber roundNumber = new RoundNumber(i);
-        while (!player.isEnded(roundNumber)) {
-            downPin(player, i);
-            outputView.printScoreBoard(player);
-        }
-    }
-
-    private PlayerName playerName() {
-        try {
-            return PlayerName.valueOf(inputView.inputPlayerName());
-        } catch (PlayerNameValidationException e) {
-            System.err.println(e.getMessage());
-            return playerName();
-        }
+        final int playerCount = inputView.inputPlayerCount();
+        final List<PlayerName> playerNames = inputView.inputPlayerNames(playerCount);
+        final BowlingGame bowlingGame = new BowlingGame(playerNames);
+        outputView.printScoreBoard(bowlingGame.scoreBoard());
+
+        final BowlingTemplate bowlingTemplate =
+                new BowlingTemplate(this::pin, () -> outputView.printScoreBoard(bowlingGame.scoreBoard()));
+        bowlingGame.playGame(bowlingTemplate);
```

Comment:
```suggestion
        Runnable printScoreBoard = printScoreBoard(bowlingGame);
        printScoreBoard.run();

        bowlingGame.playGame(new BowlingTemplate(this::pin, printScoreBoard));
```
이렇게로도 바꿔볼 수 있어 보이네요 😀 
```java
    private Runnable printScoreBoard(BowlingGame bowlingGame) {
        return () -> outputView.printScoreBoard(bowlingGame.scoreBoard());
    }
```

Reply:
아직 함수형이랑은 덜 친해서 개선포인트를 잘 못잡았네요!!
감사합니다! 리뷰어님 덕분에 많이 배울 수 있었습니다!
코딩하다가 스트레스 받기도 했었지만 리뷰로 배우는 경험때문에 좋았던 것 같아요!
저도 리뷰어님 하고 의견을 나누고 배울 수 있어 즐거운 시간이었습니다~ 감사합니다~